src/rules/max-line-length/index.js

```diff
@@ -50,7 +50,8 @@ export default function (maxLength, options) {
     }
 
     function checkNewline(match) {
-      let nextNewlineIndex = rootString.indexOf("\n", match.endIndex)
+      const endOfLineIndicator = (rootString.indexOf("\r\n", match.endIndex) !== -1) ? "\r\n" : "\n"
```

Comment:
As `checkNewline` is run on each line then I think inconsistent line endings are dealt with. However, @JasonAMartin can you please add a test to show that this is the case e.g. a short first line ending in a `\n` then a longer, rejected, line ending in `\r\n`?